piff/gp_interp_2pcf.py

```diff
@@ -0,0 +1,375 @@
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: gp_interp_2pcf
+"""
+
+import numpy as np
+import treecorr
+import copy
+
+from sklearn.gaussian_process.kernels import Kernel
+import scipy.optimize as op
+from scipy.linalg import cholesky, cho_solve
+
+from .interp import Interp
+from .star import Star, StarFit
+
+
+class GPInterp2pcf(Interp):
+    """
+    An interpolator that uses two-point correlation function and gaussian process to interpolate a single surface.
+
+    :param keys:        A list of star attributes to interpolate from
```

Comment:
well I guess it could be done for arbitrary numbers of dims, but treecorr needs it to be 1, 2, or 3, and we would need to be able to handle len(keys) != 2

Reply:
Indeed this totally right. I added a ValueError if the len(key)!=2. I added that in the documentation also.